apps/site/lib/site_web/views/event_view.ex

```diff
@@ -59,42 +61,70 @@ defmodule SiteWeb.EventView do
   @doc "Nicely renders the duration of an event, given two DateTimes."
   @spec render_event_duration(
           NaiveDateTime.t() | DateTime.t(),
-          NaiveDateTime.t() | DateTime.t() | nil
-        ) ::
+          NaiveDateTime.t() | DateTime.t() | nil,
           String.t()
-  def render_event_duration(start_time, end_time)
+        ) :: String.t() | %{date: String.t(), time: String.t()}
+  def render_event_duration(start_time, end_time, style)
 
-  def render_event_duration(start_time, nil) do
+  def render_event_duration(start_time, nil, style) do
     start_time
     |> maybe_shift_timezone
-    |> do_render_event_duration(nil)
+    |> do_render_event_duration(nil, style)
   end
 
-  def render_event_duration(start_time, end_time) do
+  def render_event_duration(start_time, end_time, style) do
     start_time
     |> maybe_shift_timezone
-    |> do_render_event_duration(maybe_shift_timezone(end_time))
+    |> do_render_event_duration(maybe_shift_timezone(end_time), style)
   end
 
-  defp do_render_event_duration(start_time, nil) do
-    "#{pretty_date(start_time, "{WDshort}, {Mshort} {D}, {YYYY}")} \u2022 #{
-      format_time(start_time)
-    }"
+  @spec do_render_event_duration(
+          NaiveDateTime.t() | DateTime.t(),
+          NaiveDateTime.t() | DateTime.t() | nil,
+          String.t()
+        ) :: String.t() | %{date: String.t(), time: String.t()}
+  defp do_render_event_duration(start_time, nil, style) do
+    if style === "list" do
+      "#{pretty_date(start_time, "{WDshort}, {Mshort} {D}, {YYYY}")} \u2022 #{
+        format_time(start_time)
+      }"
+    else
+      %{
+        date: "#{pretty_date(start_time, "{WDfull}, {Mfull} {D}, {YYYY}")}",
+        time: "#{format_time(start_time)}"
+      }
+    end
   end
 
   defp do_render_event_duration(
          %{year: year, month: month, day: day} = start_time,
-         %{year: year, month: month, day: day} = end_time
+         %{year: year, month: month, day: day} = end_time,
+         style
        ) do
-    "#{pretty_date(start_time, "{WDshort}, {Mshort} {D}, {YYYY}")} \u2022 #{
-      format_time(start_time)
-    } - #{format_time(end_time)}"
+    if style === "list" do
```

Comment:
I wonder if we should do this logic on the template side and have two 'render event' functions; as in, "if we're in the calendar view, format duration this way; otherwise, format it this other way". Given the urgency to merge this, maybe this is not that important.

Reply:
Also please remember to squash and organize all your commits.

Reply:
That's exactly what hung me up for so long yesterday -- I wanted to avoid this if-else situation, but also I couldn't figure out the elegant way to do it.  Both the list and calendar view need to go through the timezone shift in `render_event_duration`.  If I made a separate 'render event' function, then I would have had to duplicate all 5 functions being used by the list view, and I figured this was cleaner?